sql/table.go

```diff
@@ -211,8 +210,9 @@ func encodeTableKey(b []byte, v parser.Datum) ([]byte, error) {
 	case parser.DString:
 		return encoding.EncodeBytes(b, []byte(t)), nil
 	case parser.DNull:
-		// TODO(tamird,pmattis): This is a hack; we should have proper nil encoding.
-		return encoding.EncodeBytes(b, nil), nil
+		// Encode NULL as \x00primaryKeySuffix
+		// TODO(tamird/pmattis): encode more sanely
+		return append(append(b, 0), primaryIndexKeySuffix...), nil
```

Comment:
Thinking about this more, I think we should encode `NULL` as we were doing before (which wasn't correct either) and move the appending of the `primaryIndexKeySuffix` to `encodeSecondaryIndexes`. If the secondary index is not unique or it is unique but one of the columns is `NULL` we append the `primaryIndexKeySuffix`. This seems more symmetrical and avoids encoding `primaryIndexKeySuffix` multiple times into a key if multiple columns in the index key are `NULL`.